packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -126,22 +168,23 @@ export class Table extends Construct {
 
     private readonly keySchema = new Array<dynamodb.TableResource.KeySchemaProperty>();
     private readonly attributeDefinitions = new Array<dynamodb.TableResource.AttributeDefinitionProperty>();
+    private readonly globalSecondaryIndexes = new Array<dynamodb.TableResource.GlobalSecondaryIndexProperty>();
+
+    private readonly nonKeyAttributes: string[] = [];
 
     private readScalingPolicyResource?: applicationautoscaling.ScalingPolicyResource;
     private writeScalingPolicyResource?: applicationautoscaling.ScalingPolicyResource;
 
     constructor(parent: Construct, name: string, props: TableProps = {}) {
         super(parent, name);
 
-        const readCapacityUnits = props.readCapacity || 5;
-        const writeCapacityUnits = props.writeCapacity || 5;
-
         this.table = new dynamodb.TableResource(this, 'Resource', {
             tableName: props.tableName,
             keySchema: this.keySchema,
             attributeDefinitions: this.attributeDefinitions,
+            globalSecondaryIndexes: this.globalSecondaryIndexes,
```

Comment:
If there are no GSIs, can this array be empty or do we have to use the value `undefined` (leave it out in CloudFormation?). Different resources have different rules here. I'd like to see a table without a GSI in the integ test to confirm that an empty array is okay.

Reply:
An empty array works.

I agree with having an integration test. I will add table only case.